pytube/metadata.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+"""This module contains the YouTubeMetadata class."""
+import json
+from typing import Dict
+from typing import List
+from typing import Optional
+
+
+class YouTubeMetadata:
+    def __init__(self, metadata: List):
```

Comment:
instead of list we can take iterator so we don't have to convert around that much, but isnt particularly important

Reply:
I don't think we should take an iterator here, because the code for building the object expects a list. The metadata itself is a list of json objects, all that I'm doing in `extract.metadata` is filtering some of those elements out:
![image](https://user-images.githubusercontent.com/9667869/98960957-06759280-24d3-11eb-978b-5a8bde963ede.png)
This __init__ implementation strictly will not work with certain types of iterators, so I believe the argument should be as specific as possible, in this case a List.